lib/FrontendTool/FrontendTool.cpp

```diff
@@ -393,7 +393,7 @@ LLVM_ATTRIBUTE_NOINLINE
 static void debugFailWithAssertion() {
   // This assertion should always fail, per the user's request, and should
   // not be converted to llvm_unreachable.
-  assert(0 && "This is an assertion!");
+  assert((0) && "This is an assertion!");
```

Comment:
llvm_unreachable is compiled in the same cases as asserts, so there's no point in not making this llvm_unreachable

Reply:
Presumably there's a reason why this specifically says it should not be llvm_unreachable.

Reply:
Yeah, a misunderstanding on the author's part :)

Reply:
Well, the original patch did this:

```
commit 9df53c5c71b5bb9252ee5d13cd349295a6d19602
Author: Connor Wakamo <cwakamo@apple.com>
Date:   Fri Feb 7 22:30:40 2014 +0000

    [frontend] Switch -debug-assert-* from llvm_unreachable() to assert(0).
    
    Swift SVN r13656
```

Reply:
The flag is only used in one test, test/Driver/assert.swift. If the test passes with it changed to an llvm_unreachable I would keep it that way.

Reply:
llvm_unreachable actually has an effect in no-asserts build (which is why I never introduce it, and almost certainly why it was not used above).

It becomes __builtin_unreachable() which the optimizer takes to mean "party on!", which in practice for LLVM I believe means it deletes all code following this point until the next join in control flow.